source/tap/tracer.go

```diff
@@ -36,8 +36,58 @@ const (
 )
 
 // Tracer implements a gwr data source that allows easy tracing of scope data,
-// such as function calls; another useful use case for go programs is tracing
-// each work unit handled by a worker goroutine
+// such as function calls, or rounds of a worker goroutine's loop.
+//
+// Tracers should be created for each area of the application that can be
+// traced.  This could be as simple as creating a package-level tracer:
+//
+//     package foo
+//
+//     import "github.com/uber-go/gwr/source"
+//
+//     tracer := source.AddNewTracer("foo")
+//
+// Tracers can also be attached to parts of the application:
+//
+//     type Thing struct {
+//         t *Tracer
+//     }
+//
+//     func NewThing() *Thing {
+//         // ...
+//         t.tracer = source.AddNewTracer(fmt.Sprintf("foo/%v", someThingIdentifier))
+//         // ...
+//     }
+//
+// If Things are not the same life-cycle as the application, then they should
+// have teardown code to remove their tracer data sources:
+//
+//     gwr.DefaultDataSources.Remove(t.tracer.Name())
+//
+// You can then proceed to trace your functions and methods:
+// 1. Decide where/what you want to start tracing; this will probably be one or
+//    more exported functions or methods called by your user.
+// 2. Within these entry points use Tracer.Scope to create a root scope.  Call
+//    scope.OpenCall to note the start of function, then defer a, or manually
```

Comment:
`then defer,` or might be better to say `close the scope via closecall, error, errorname`